### ISSUE TITLE
Warning about unused import in test module despite being used in a test

### ISSUE THREAD

**Issue by camelid** [rust-lang/rust#72522]
Rust is warning about an unused import even though the import is used.

This issue shows up with Clippy as well. Is this the right place to report it?

Rust also warned about `foo` being unused, but it's only used in a test, so I figure that's correct.

I made a playground for it:

```rust
fn foo() {}

mod test {
    use super::*;
    
    #[test]
    fn bar() {
        assert_eq!(foo(), ())
    }
}
```

([Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=88e2efcbef5d4c81ea766edd3d285c4b))

Output:

```

running 1 test
test test::bar ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out


running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out


```

Errors:

```
   Compiling playground v0.0.1 (/playground)
warning: unused import: `super::*`
 --> src/lib.rs:4:9
  |
4 |     use super::*;
  |         ^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: function is never used: `foo`
 --> src/lib.rs:1:4
  |
1 | fn foo() {}
  |    ^^^
  |
  = note: `#[warn(dead_code)]` on by default

    Finished test [unoptimized + debuginfo] target(s) in 0.67s
     Running target/debug/deps/playground-62708a348f5d0345
   Doc-tests playground

```

**Comment by doctorn** [comment]
You should add the `#[cfg(test)]` annotation to the module.

**Comment by camelid** [comment]
Ah, thank you! Sorry about filing a false bug...
